angular-oauth2-oidc/src/oauth-service.ts

```diff
@@ -1085,8 +1141,38 @@ export class OAuthService
 
         options = options || { };
 
-        let parts: object;
+        if (!this.requestAccessToken && !this.oidc) {
+            return Promise.reject('Either requestAccessToken or oidc or both must be true.');
+        }
+
+        if (window.location.search && window.location.search.startsWith('?code')) {
+            return this.tryLoginAuthorizationCode();
+        } else {
+            return this.tryLoginImplicit(options);
+        }
+    };
+
+    private tryLoginAuthorizationCode(): Promise<void> {
```

Comment:
@bechhansen @manfredsteyer - this throws a ngc compile error `Promise<any>` fixes, but might not be what is intended, you can do or cases, but we probably resolve with the code?